src/coord/src/coord.rs

```diff
@@ -2668,22 +2636,74 @@ impl Coordinator {
         plan: DropComputeInstancesPlan,
     ) -> Result<ExecuteResponse, CoordError> {
         let mut ops = Vec::new();
-        let mut instance_ids = Vec::new();
+        let mut instance_replica_drop_sets = Vec::with_capacity(plan.names.len());
         for name in plan.names {
             let instance = self.catalog.resolve_compute_instance(&name)?;
-            instance_ids.push(instance.id);
+            instance_replica_drop_sets.push((
+                instance.id,
+                instance.name.clone(),
+                instance.replicas.clone(),
+            ));
+            for replica_name in instance.replicas.keys() {
+                ops.push(catalog::Op::DropComputeInstanceReplica {
+                    name: replica_name.to_string(),
+                    compute_id: instance.id,
+                });
+            }
             let ids_to_drop: Vec<GlobalId> = instance.indexes().iter().cloned().collect();
             ops.extend(self.catalog.drop_items_ops(&ids_to_drop));
             ops.push(catalog::Op::DropComputeInstance { name });
         }
 
         self.catalog_transact(ops, |_| Ok(())).await?;
-        for id in instance_ids {
-            self.dataflow_client.drop_instance(id).await.unwrap();
+        for (instance_id, instance_name, replicas) in instance_replica_drop_sets {
+            for (replica_name, config) in replicas {
+                self.dataflow_client
+                    .drop_replica(instance_id, &instance_name, &replica_name, config)
+                    .await
+                    .unwrap();
+            }
+            self.dataflow_client
+                .drop_instance(instance_id)
+                .await
+                .unwrap();
         }
+
         Ok(ExecuteResponse::DroppedComputeInstance)
     }
 
+    async fn sequence_drop_compute_instance_replica(
+        &mut self,
+        DropComputeInstanceReplicaPlan { names, cluster }: DropComputeInstanceReplicaPlan,
+    ) -> Result<ExecuteResponse, CoordError> {
+        if names.is_empty() {
+            return Ok(ExecuteResponse::DroppedComputeInstanceReplicas);
+        }
+        let cluster_name = cluster.unwrap();
```

Comment:
This is subjective so feel free to ignore. It might make more sense for `DropComputeInstanceReplicaPlan` to be `Option<(Vec<String>, String)`, so you don't accidentally end up in a situation where `cluster` is `None`, but `names` is not empty.

Reply:
If names is not empty, cluster being none is a programming error and should panic, which is what the unwrap does--not quite sure what the difference is here?

Reply:
Yeah sorry, I don't really think there is a practical difference. I was just thinking that `Option<(Vec<String>, String)` might make it harder for future programmers to make that error, though it's pretty subjective. I think it's fine the way it is.